File: ios/Classes/Bridge/SaveDecisionsBridge.swift
import Usercentrics

struct SaveDecisionsBridge : MethodBridge {

    let name: String = "saveDecisions"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let decisions = (argsDict["decisions"] as! [Any]).map { UserDecision(from: $0 as! NSDictionary)}
        let consentType = UsercentricsConsentType.initialize(from: argsDict["consentType"])!
        let consents = usercentrics.shared.saveDecisions(decisions: decisions, consentType: consentType)
        result(consents.map { $0.serialize() })
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/SaveDecisionsBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.deserializeUserDecision
import com.usercentrics.sdk.flutter.serializer.serialize
import com.usercentrics.sdk.models.settings.UsercentricsConsentType

internal class SaveDecisionsBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "saveDecisions"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        val argsMap = call.arguments as Map<*, *>
        val consents = usercentrics.instance.saveDecisions(
            decisions = (argsMap["decisions"] as List<*>).map { it.deserializeUserDecision() },
            consentType = UsercentricsConsentType.valueOf(argsMap["consentType"] as String)
        )
        result.success(consents.map { it.serialize() })
    }

}